// timestamp-date/service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimestampDate = void 0;
const timestamp_1 = require("./actions/timestamp");
const string_1 = require("./actions/string");
const cloneDeep = require("lodash.clonedeep");
require("firebase/app");
class TimestampDate {
    constructor() {
        this.tsHandler = new timestamp_1.TimestampHandler();
        this.strHandler = new string_1.StringDateHandler();
    }
    getServerTimestamp() {
        return this.tsHandler.getServerTimestamp();
    }
    getServerDate() {
        return this.getServerTimestamp().toDate();
    }
    timestampToDate(timestamp) {
        return this.tsHandler.timestampToDate(timestamp);
    }
    dateToTimestamp(date) {
        return this.tsHandler.dateToTimestamp(date);
    }
    parseTimestampToDate(data) {
        return this.tsHandler.parseTimestampToDate(cloneDeep(data));
    }
    parseDateToTimestamp(data) {
        return this.tsHandler.parseDateToTimestamp(cloneDeep(data));
    }
    parseStringToDate(data) {
        return this.strHandler.parseStringToDate(cloneDeep(data));
    }
    parseDateToString(data) {
        return this.strHandler.parseDateToString(cloneDeep(data));
    }
}
exports.TimestampDate = TimestampDate;
